GroversAlgorithm/GroversAlgorithm.ipynb

```diff
@@ -288,14 +288,16 @@
     "\n",
     " * If the register is in any other basis state, multiply its phase by -1.\n",
     "\n",
-    "> This operation implements operator $2|0...0\\rangle\\langle0...0| - I$.\n",
+    "> This operation implements operator $2|0...0\\rangle\\langle0...0| - I$ $ = \\left(\\begin{matrix}1&0&…&0\\\\0&-1&…&0\\\\\\vdots&\\vdots&\\vdots&\\vdots\\\\0&0&0&-1\\end{matrix}\\right) $\n",
     "\n",
     "<br/>\n",
     "<details>\n",
     "  <summary>Need a hint? Click here </summary>\n",
     "    Note that quantum states are defined up to a global phase.\n",
     "    Thus the state obtained as a result of this operation is the same\n",
-    "    as the state obtained by flipping the sign of only the $|0...0\\rangle$ state.\n",
+    "    as the state obtained by flipping the sign of only the $|0...0\\rangle$ state and ignoring the global phase -1.<br>\n",
+    "    $-\\big(2|0...0\\rangle\\langle0...0| - I\\big) = I - 2|0...0\\rangle\\langle0...0| = \\left(\\begin{matrix}-1&0&…&0\\\\0&1&…&0\\\\\\vdots&\\vdots&\\vdots&\\vdots\\\\0&0&0&1\\end{matrix}\\right) $<br>\n",
+    "    This can be implemented as $Controlled Z(|s_0 s_1 \\ldots s_{n-2}\\rangle, |s_{n-1}\\rangle)$ which results in: $|1...10\\rangle \\rightarrow |1...10\\rangle, |1...11\\rangle \\rightarrow -|1...11\\rangle$ due to $Z(|0\\rangle)=|0\\rangle$ and $Z(|1\\rangle)=-|1\\rangle$ if you flip all bits (apply $X$) before and after the operation.  \n",
```

Comment:
I would move this part of the hint to the 2nd slot - it's more about *how* to implement this unitary than about *what unitary* to implement, so taking one hint with both parts takes the learner too far, and we want to give them just a little nudge in the right direction. In the 2nd hint you'll be able to give this suggestion, and modify the existing hint to say "Alternatively, you can use the same trick..."

I would also leave out the last detail, suggesting to use Controlled Z but not spelling out that you need to flip all the bits. Something like "Consider the Controlled Z gate, applied with most of the qubits as control and the last qubit as target: (your explanation here up to $Z(|1\\rangle)=-|1\\rangle$)".